suggested in http://projects.scipy.org/numpy/ticket/1818 but not accepted 
upstream due to the policy described in http://projects.scipy.org/numpy/ticket/573
--- ./numpy/distutils/fcompiler/gnu.py	2011-09-14 05:39:16.000000000 +0900
+++ /home/benda/tmp/gnu.py	2012-09-27 01:07:07.320690000 +0900
@@ -23,6 +23,8 @@
 if is_win64():
     #_EXTRAFLAGS = ["-fno-leading-underscore"]
     _EXTRAFLAGS = []
+elif os.uname()[4] == 'x86_64':
+    _EXTRAFLAGS = ['-fPIC']
 else:
     _EXTRAFLAGS = []
 
@@ -255,7 +257,7 @@
         'compiler_f90' : [None, "-Wall", "-fno-second-underscore"] + _EXTRAFLAGS,
         'compiler_fix' : [None, "-Wall", "-ffixed-form",
                           "-fno-second-underscore"] + _EXTRAFLAGS,
-        'linker_so'    : ["<F90>", "-Wall"],
+        'linker_so'    : ["<F90>", "-Wall", "-shared"],
         'archiver'     : ["ar", "-cr"],
         'ranlib'       : ["ranlib"],
         'linker_exe'   : [None, "-Wall"]
